lib/Webhooks.js

```diff
@@ -104,7 +99,8 @@ const signature = {
       });
     }
 
-    const expectedSignature = this._computeSignature(
+    cryptoProvider = cryptoProvider || getNodeCryptoProvider();
+    const expectedSignature = cryptoProvider.computeHMACSignature(
```

Comment:
This might be evil, but for async support:
```diff
- const expectedSignature = cryptoProvider.computeHMACSignature(
-    ...
- );
- // Do a bunch of stuff with expectedSignature
+ const compare = (expectedSignature) => {
+   // Do a bunch of stuff with expectedSignature
+ };
+ const expectedSignature = cryptoProvider.computeHMACSignature(
+   ...
+ );
+ if (expectedSignature && typeof expectedSignature.then === 'function') {
+   return expectedSignature.then(compare);
+ } else {
+   return compare(expectedSignature);
+ }
```

Reply:
Interesting. I think this could potentially simplify the implementation over having two separate methods... let's think about it and see how the need comes up? Either change would require updating the types though I imagine changing `constructEvent`'s return type from `Stripe.Event` to `Stripe.Event | Promise<Stripe.Event>` wouldn't technically be backwards compatible? 

Think we can merge this and handle async in a separate PR.